environment-pipelines/locals.tf

```diff
@@ -21,16 +21,23 @@ locals {
 
   # Convert the env config into a list and add env name and vpc / requires_approval from the environments config.
-  environment_config        = [for name, env in var.environments : merge(lookup(local.base_env_config, name, {}), env, { "name" = name })]
-  triggers_another_pipeline = var.pipeline_to_trigger != null
+  environment_config = [for name, env in var.environments : merge(lookup(local.base_env_config, name, {}), env, { "name" = name })]
 
-  triggered_pipeline_account_name = local.triggers_another_pipeline ? var.all_pipelines[var.pipeline_to_trigger].account : null
-  triggered_account_id            = local.triggers_another_pipeline ? local.account_map[local.triggered_pipeline_account_name] : null
+  triggers_another_pipeline         = var.pipeline_to_trigger != null
+  triggered_pipeline_account_name   = local.triggers_another_pipeline ? var.all_pipelines[var.pipeline_to_trigger].account : null
+  triggered_account_id              = local.triggers_another_pipeline ? local.account_map[local.triggered_pipeline_account_name] : null
+  triggered_pipeline_codebuild_role = local.triggers_another_pipeline ? "arn:aws:iam::${local.triggered_account_id}:role/${var.application}-${var.pipeline_to_trigger}-environment-pipeline-codebuild" : null
+  triggered_pipeline_environments   = local.triggers_another_pipeline ? [for name, config in var.all_pipelines[var.pipeline_to_trigger].environments : { "name" = name }] : null
 
   list_of_triggering_pipelines     = [for pipeline, config in var.all_pipelines : merge(config, { name = pipeline }) if lookup(config, "pipeline_to_trigger", null) == var.pipeline_name]
   set_of_triggering_pipeline_names = toset([for pipeline in local.list_of_triggering_pipelines : pipeline.name])
+  triggering_pipeline_role_arns    = [for name in local.set_of_triggering_pipeline_names : "arn:aws:iam::${local.account_map[var.all_pipelines[name].account]}:role/${var.application}-${name}-environment-pipeline-codebuild"]
 
-  triggering_pipeline_role_arns = [for name in local.set_of_triggering_pipeline_names : "arn:aws:iam::${local.account_map[var.all_pipelines[name].account]}:role/${var.application}-${name}-environment-pipeline-codebuild"]
+  triggered_by_another_pipeline      = length([for config in var.all_pipelines : true if lookup(config, "pipeline_to_trigger", null) == var.pipeline_name]) > 0
+  triggering_pipeline_account_name   = local.triggered_by_another_pipeline ? one(local.list_of_triggering_pipelines).account : null
+  triggering_account_id              = local.triggered_by_another_pipeline ? local.account_map[local.triggering_pipeline_account_name] : null
+  triggering_pipeline_name           = local.triggered_by_another_pipeline ? one(local.list_of_triggering_pipelines).name : null
+  triggering_pipeline_codebuild_role = local.triggered_by_another_pipeline ? "arn:aws:iam::${local.triggering_account_id}:role/${var.application}-${local.triggering_pipeline_name}-environment-pipeline-codebuild" : null
```

Comment:
Please could you comment to highlight the actual change(s) above.

Reply:
The formatting has made that horrible to read. This won't be much better but it's adding these lines:

`triggered_pipeline_codebuild_role = local.triggers_another_pipeline ? "arn:aws:iam::${local.triggered_account_id}:role/${var.application}-${var.pipeline_to_trigger}-environment-pipeline-codebuild" : null`

`triggered_pipeline_environments = local.triggers_another_pipeline ? [for name, config in var.all_pipelines[var.pipeline_to_trigger].environments : { "name" = name }] : null`

`triggered_by_another_pipeline = length([for config in var.all_pipelines : true if lookup(config, "pipeline_to_trigger", null) == var.pipeline_name]) > 0`

`triggering_pipeline_account_name = local.triggered_by_another_pipeline ? one(local.list_of_triggering_pipelines).account : null`

`triggering_account_id = local.triggered_by_another_pipeline ? local.account_map[local.triggering_pipeline_account_name] : null`

`triggering_pipeline_name = local.triggered_by_another_pipeline ? one(local.list_of_triggering_pipelines).name : null`

`triggering_pipeline_codebuild_role = local.triggered_by_another_pipeline ? "arn:aws:iam::${local.triggering_account_id}:role/${var.application}-${local.triggering_pipeline_name}-environment-pipeline-codebuild" : null`